pkg/internal/resultref/resultref.go

```diff
@@ -42,7 +42,7 @@ const (
 
 	// arrayIndexing will match all `[int]` and `[*]` for parseExpression
 	arrayIndexing          = `\[([0-9])*\*?\]`
-	stepResultUsagePattern = `\$\(steps\..*\.results\..*\)`
+	stepResultUsagePattern = `\$\(steps\..*?\.results\..*?\)`
```

Comment:
I'll revert this.. this should be fixed in a separate issue https://github.com/tektoncd/pipeline/issues/7764